src/vi/vlogtruyen/src/eu/kanade/tachiyomi/extension/vi/vlogtruyen/VlogTruyen.kt

```diff
@@ -78,60 +80,56 @@ class VlogTruyen : ParsedHttpSource(), ConfigurableSource {
         status = parseStatus(document.selectFirst("div.top-detail-manga > div.top-detail-manga-avatar > div.manga-status > p")?.text())
         author = document.select(".h5-drawer:contains(Tác Giả) + ul li a").joinToString { it.text() }
     }
 
     private fun parseStatus(status: String?) = when {
         status == null -> SManga.UNKNOWN
-        status.contains("Đang tiến hành") -> SManga.ONGOING
-        status.contains("Đã hoàn thành") -> SManga.COMPLETED
-        status.contains("Tạm ngưng") -> SManga.ON_HIATUS
+        status.contains("Đang tiến hành", ignoreCase = true) -> SManga.ONGOING
+        status.contains("Đã hoàn thành", ignoreCase = true) -> SManga.COMPLETED
+        status.contains("Tạm ngưng", ignoreCase = true) -> SManga.ON_HIATUS
         else -> SManga.UNKNOWN
     }
 
+    override fun chapterListRequest(manga: SManga): Request {
+        val slug = manga.url.substringAfterLast("/").substringBeforeLast(".")
+        return GET("$baseUrl/thong-tin-ca-nhan?manga_slug=$slug", xhrHeaders)
+    }
+
     override fun chapterListParse(response: Response): List<SChapter> {
         val json = response.parseAs<ChapterDTO>()
         val document = Jsoup.parseBodyFragment(json.data.chaptersHtml, response.request.url.toString())
         val hidePaidChapters = preferences.getBoolean(KEY_HIDE_PAID_CHAPTERS, false)
-        return document.select("li").filterNot {
+        return document.select(chapterListSelector()).filterNot {
             if (hidePaidChapters) {
                 it.select("li > b").isNotEmpty()
             } else {
                 false
             }
         }
-            .mapNotNull {
-                SChapter.create().apply {
-                    setUrlWithoutDomain(it.selectFirst("a")!!.attr("href"))
-                    name = it.select("h3").first()!!.text().trim()
-                    if (it.select("li > b").text().isNotBlank()) {
-                        name += " " + it.select("li > b").text() + " 🔒"
-                    }
-                    date_upload = dateFormat.tryParse(it.select("li:not(:has(> span.chapter-view)) > span, li > span:last-child").text())
-                }
-            }
+            .map { element -> chapterFromElement(element) }
     }
 
-    override fun fetchChapterList(manga: SManga): Observable<List<SChapter>> {
-        val url = client.newCall(GET(baseUrl + manga.url, headers)).execute().asJsoup()
-        if (checkChapterLists(url).isNotEmpty()) {
-            val mangaId = checkChapterLists(url)
-            return client.newCall(GET("$baseUrl/thong-tin-ca-nhan?manga_id=$mangaId", headers))
-                .asObservableSuccess()
-                .map { response -> chapterListParse(response) }
+    override fun chapterListSelector() = "li"
+
+    override fun chapterFromElement(element: Element): SChapter = SChapter.create().apply {
+        setUrlWithoutDomain(element.selectFirst("a")!!.absUrl("href"))
+        name = element.select("h3").first()!!.text().trim()
```

Comment:
`.text()` is already trimmed.

```suggestion
        name = element.select("h3").first()!!.text()
```